hax/hax/motr/__init__.py

```diff
@@ -250,17 +248,15 @@ def send_entrypoint_request_reply(self, message: EntrypointRequest):
                     rm_eps = svc.address
                     break
             if confds and (not self.is_stopping) and (not rm_eps):
-                if util.m0ds_stopping():
-                    e_rc = 0
                 raise RuntimeError('No RM node found in Consul')
         except Exception:
+            # In case of exception we are dropping the entrypoint requests
+            # silently at hax and hax will not send any entrypoint reply.
+            # In such a case we will let motr timeout and Hare will then
+            # expect a subsequent entrypoint request from Motr
```

Comment:
Doesn't it actually mean that mkfs does not handle `EAGAIN` properly?

Reply:
Yes, to summarize, mkfs does not handle EAGAIN properly
cc @mssawant

Reply:
Then it's a workaround. Does motr team know about this issue? Do they have plans to fix it in a right way?

I don't like the idea of collecting workarounds. If we do have to put them, we have to have plans how they get resolved in the most straight way.

Reply:
Yes motr team is aware of it.
Discussion is done at https://jts.seagate.com/browse/EOS-25926?focusedCommentId=2263543&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2263543

Reply:
Yes, even I am not very happy with such an approach, but better than 1 sec delay. This was discussed with Motr team and they presently don't have a fix at their end.

Reply:
/**
 * Command passed to the entrypoint consumer side to control entrypoint request
 * flow. Producer side may request consumer to keep querying entrypoint, or
 * stop querying and force process to quit.
 */
enum m0_ha_entrypoint_control {
        M0_HA_ENTRYPOINT_CONSUME,      /**< Entrypoint is good to be consumed */
        M0_HA_ENTRYPOINT_QUIT,         /**< Remote consumer requested to quit */
        M0_HA_ENTRYPOINT_QUERY,        /**< Remote consumer must query again */
#if 0   /*
         * XXX: Possible way to introduce explicit delay. The entry implies the
         * value to be transformed to M0_HA_ENTRYPOINT_QUERY immediately in the
         * interface API m0_halon_interface_entrypoint_reply() along with
         * setting delay value correspondingly.
         */
        M0_HA_ENTRYPOINT_QUERY_1SEC,   /**< Query again, with 1 sec delay */
        /*
         * Other ways to introduce the delay between HA client queries:
         * - add explicit delay_ns parameter to the Halon interface API
         *   m0_halon_interface_entrypoint_reply(), or
         * - synthesize delay value on HA client side based on some policy
         *   (predefined constant, or algorithmic, or statistics-based), or
         * - transform control values like M0_HA_ENTRYPOINT_QUERY_1SEC to
         *   M0_HA_ENTRYPOINT_QUERY and delay on HA client side.
         */
#endif
};
From ha/entrypoint_fops.h

@mssawant : I have created the motr issue https://jts.seagate.com/browse/EOS-27068 to solve it properly if the options in hare are only short term fixes..

Reply:
@SwapnilGaonkar7, please add the above ticket to the comment.

Reply:
Done